main.cpp

```diff
@@ -1,81 +1,127 @@
+    std::cout << character1.GetName() << " wins. Remaining HP:" << character1.GetHealth() << '\n';
   }
   else{
-    std::cout << character1.GetName() << " died. " << character2.GetName() << " wins.\n";
+    std::cout << character2.GetName() << " wins. Remaining HP:" << character2.GetHealth() << '\n';
   }
 }
 
-int checkHealthOrDamageInputArgument(char *argument){
-  int number;
-  try {
-    number = std::atoi(argument);
-  } catch (...) {}
+std::vector<std::string> splittedString(std::string text, char delimiter){
+  std::vector<std::string> splittedStrings;
+  std::string addString = "";
 
-  return number;
-}
-
-int main(int argc, char *argv[])
-{
-  if(argc != 7){
-    std::cout << "Something is wrong with the arguments.\nExample: Maple 150 10 Sally 45 30\n";
-    return 1;
-  }
-  else{
-    std::string character1_name(argv[1]);
-
-    int character1_health = checkHealthOrDamageInputArgument(argv[2]);
-    if(character1_health == 0){
-      std::cout << argv[2] << " is not valid for health\n";
-      return 1;
+  for(unsigned int i = 0; i < text.length(); i++){
+    if(text[i] == delimiter){
+      splittedStrings.push_back((addString));
+      addString = "";
     }
-
-    int character1_damage = checkHealthOrDamageInputArgument(argv[3]);
-    if(character1_damage == 0){
-      std::cout << argv[3] << " is not valid for damage\n";
-      return 1;
+    else{
+      addString += text[i];
     }
+  }
 
-    Character character1(character1_name, character1_health, character1_damage);
+  splittedStrings.push_back((addString));
 
-    std::string character2_name(argv[4]);
+  return splittedStrings;
+}
 
-    int character2_health = checkHealthOrDamageInputArgument(argv[5]);
-    if(character2_health == 0){
-      std::cout << argv[5] << " is not valid for health\n";
-      return 1;
-    }
+bool fileExists(const std::string& fileName)
+{
+	std::ifstream f(fileName.c_str());
+	return f.good();
+}
 
-    int character2_damage = checkHealthOrDamageInputArgument(argv[6]);
-    if(character2_damage == 0){
-      std::cout << argv[6] << " is not valid for damage\n";
-      return 1;
+static Character* parseUnit(const std::string& fileName){
+  std::string name = "";
+  int health = 0;
+  int damage = 0;
+
+  std::fstream characterDataFile;
+  characterDataFile.open(fileName);
+ 
+  std::string line;
+  int lineIndex = 0;
+  if(characterDataFile.is_open()){
+    while (std::getline(characterDataFile, line))
+    {
+      if(lineIndex == 1){
+        name = splittedString(line, ':').back();
+        name = name.substr(2, name.length() - 4);
+      }
+      else if(lineIndex == 2){
+        std::string inputHealth =  splittedString(line, ':').back();
+        inputHealth = inputHealth.substr(0, inputHealth.length()-1);
+        health = std::stoi(inputHealth);
+      }
+      else if(lineIndex == 3){
+        std::string damageHealth =  splittedString(line, ':').back();
+        damage = std::stoi(damageHealth);
+      }
+      lineIndex++;
```

Comment:
Rendben van, de amugy ha mar megcsinaltatok ezt a splitted dolgot, akkor nem lenne nagy melo tetszoleges sorrendben beolvasni, csak elobb splittelsz, aztan nezed az elsot.

Reply:
Nem egészen értem, mit értesz az alatt hogy 

>  nem lenne nagy melo tetszoleges sorrendben beolvasni, csak elobb splittelsz, aztan nezed az elsot.

Vagy azt hogy olvassuk ki a 2., 3. és 4. sort csak úgy?

Reply:
Hat hogy eloszor csinalsz egy split-et, amibol kivolvasod mi a key, azutan csinalsz egy if-et, es allitod be a megfelelo valtozot.

Reply:
ahaa, értem. köszi 😁